contracts/eosiolib/privileged.h

```diff
@@ -29,19 +37,56 @@ extern "C" {
     */
    int64_t set_proposed_producers( char *producer_data, uint32_t producer_data_size );
 
+   /**
+    * @brief Set new active producers
+    * Set new active producers. Producers will only be activated once the block which starts the next round is irrreversible
```

Comment:
The conditions for transition are out of date, they are now:
calling this method "proposes" a schedule change
once the block that contains the proposal becomes irreversible, the schedule is promoted to "pending" automatically.  Once the block that promotes the schedule is irreversible, the schedule will become "active" 

It is no longer based on rounds etc.